contracts/marketplace/nft_marketplace.ral

```diff
@@ -57,38 +57,42 @@ Contract NFTMarketPlace(
         let nftListingContractId = subContractId!(tokenId)
         let nftListing = NFTListing(nftListingContractId)
         let buyer = callerAddress!()
-        let price = nftListing.getPrice()
         let tokenOwner = nftListing.getTokenOwner()
+        let price = nftListing.getPrice()
+        let commissionRateOfListing = nftListing.getCommissionRate()
+        let listingFeeOfListing = nftListing.getListingFee()
 
-        nftListing.buy{buyer -> ALPH: price}(buyer)
+        let priceAfterFee = checkListingPriceAfterFee(price, commissionRateOfListing, listingFeeOfListing)
```

Comment:
Question:

1. If someone lists an NFT, then the marketplace updates the commission rate, causing the `checkListingPriceAfterFee` fails, does this means that the NFT seller has to update the price?
2. Do we also need to call `checkListingPriceAfterFee` in the `updateNFTPrice` function?

Reply:
Sorry, please ignore the first question, I found that we saved the previous commission rate and listing price in the `NFTListing` contract.

Reply:
Sounds great :+1: